groovy/postInit/chemistry/organic_chemistry/polymers/thermosets/RubberChain.groovy

```diff
@@ -159,70 +110,98 @@ def shapes = [
-                .duration(rubber.duration)
-                .EUt(7)
-                .buildAndRegister()
-            }
-            for (catalyser in catalysers) {
-                if(rubber.isFluid)  {
-                    VULCANIZING_RECIPES.recipeBuilder()
-                    .fluidInputs(fluid(rubber.name) * rubber.amount_required * 1000)
-                    .inputs(ore(sulfurSource.name) * sulfurSource.amount_required)
-                    .notConsumable(metaitem('shape.extruder.' + shape.shapeName))
-                    .notConsumable(ore(catalyser.name))
-                    .outputs(ore(shape.name + rubber.output)[0] * (rubber.yield * shape.yield))
-                    .duration(rubber.duration.intdiv(catalyser.speed_bonus))
-                    .EUt(7)
-                    .buildAndRegister()
-                } 
-                else {
-                    VULCANIZING_RECIPES.recipeBuilder()
-                    .inputs(ore(rubber.name) * rubber.amount_required)
-                    .inputs(ore(sulfurSource.name) * sulfurSource.amount_required)
+    for (sulfur in sulfurs) { 
+        for (catalyst in catalysts) {
+            for (shape in shapes) {
+                if (oreDict.getItems(shape.name + rubber.output).size() == 0) {
+                    continue;
+                }
+                VULCANIZER.recipeBuilder()
+                    .tap {
+                        rubber instanceof Fluid ?
+                            fluidInputs(fluid(rubber.name) * (rubber.amount_required * 1000)) :
+                            inputs(ore(rubber.name) * rubber.amount_required)
+                    }
+                    .tap {
+                        sulfur instanceof Fluid ?
+                            fluidInputs(fluid(sulfur.name) * (sulfur.amount_required * 1000)) :
+                            inputs(ore(sulfur.name) * sulfur.amount_required)
+                    }
+                    .tap {
+                        if (catalyst.speed_bonus == 0) {
+                            circuitMeta(2)
+                            duration(rubber.duration)
+                        } else {
+                            notConsumable(ore(catalyst.name))
+                            duration(rubber.duration.intdiv(catalyst.speed_bonus))
+                        }
+                    }
                     .notConsumable(metaitem('shape.extruder.' + shape.shapeName))
-                    .notConsumable(ore(catalyser.name))
                     .outputs(ore(shape.name + rubber.output)[0] * (rubber.yield * shape.yield))
-                    .duration(rubber.duration.intdiv(catalyser.speed_bonus))
-                    .EUt(7)
+                    .EUt(Globals.voltAmps[0])
                     .buildAndRegister()
-                }
             }
         }
     }
 }
 
-CoagulationRecipe(1, 150)
+// Ebonite
+for (rubber in rubbers) {
+    coal_amount = (rubber.amount_required > 4) ? rubber.amount_required.intdiv(4) : 1
```

Comment:
you can use Math.max for this

Reply:
then intdiv would be called every time but ok

Reply:
wdym when you say it would be called everytime

Reply:
here it's called only if `rubber.amount_required > 4` (should be `>=8` though) but `Math.max(1, rubber.amount_required.intdiv(4))` will perform division regardless of `amount_required`

nvm i already changed to `Math.max`